Tests/Objects/IGTestDiffingObject.m

```diff
@@ -28,8 +28,18 @@ - (NSString *)description {
     return self.key;
 }
 
-- (BOOL)isEqualToDiffableObject:(id<IGListDiffable>)object {
-    return YES;
+- (BOOL)isEqualToDiffableObject:(id)object {
+    if (object == self) {
+        return YES;
+    }
+    if ([object isKindOfClass:[IGTestDiffingObject class]]) {
+        /* A simple equality test that only looks at the number of objects for the key.
+           It does not currently test the equality of each of the objects. */
+        IGTestDiffingObject *testDiffingObject = (IGTestDiffingObject *)object;
+        return self.objects.count == testDiffingObject.objects.count;
+    }
+    
+    return NO;
```

Comment:
Noting to myself that this actually tests not tripping the assert anymore. [See comment](https://github.com/Instagram/IGListKit/issues/1174#issuecomment-403966485)

Reply:
Good idea. Pull request has been updated @rnystrom.